packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -146,26 +146,17 @@ const ElementHook: Hooks<VElement> = {
     },
 };
 
-function isUpgradableComponent(
-    ctor: CustomElementConstructor | UpgradableCustomElementConstructor
-): ctor is UpgradableCustomElementConstructor {
-    return 'upgrade' in ctor;
-}
-
 const CustomElementHook: Hooks<VCustomElement> = {
     create: (vnode) => {
         const {
             sel,
             owner: { renderer },
         } = vnode;
         const UpgradableConstructor = getUpgradableConstructor(sel, renderer);
-        if (isUpgradableComponent(UpgradableConstructor)) {
+        const elm = new UpgradableConstructor((elm: HTMLElement) => {
```

Comment:
This might be an issue if the existing custom element expects parameters of a certain type. This is definitely an anti-pattern, so I think we should be able to live with it.

```js
class XImage extends HTMLElement {
  constructor(src) {
    super();
    
    if (src !== undefined && typeof src !== 'string') {
    	throw new Error('Invalid first argument, expected a string');
    }
  }
}

customElements.define('x-image', XImage);

document.createElement('x-image'); // OK
new XImage(); // OK
new XImage(() => {}); // Not OK
```

Reply:
I'm not sure what the question or suggestion is here, can you elaborate more?

Reply:
This PR changes the way we invoke the custom element constructor by passing a new argument. This extra argument will be passed to the upgradable elements and also to the custom elements that are already registered on the global CE registry.

With this approach, LWC will not be able to create standard custom elements that are doing argument checks in the constructor.  The `x-image` can't be instantiated by the LWC engine because the engine will pass a function as first argument while `XImage` accepts a string or undefined as first argument.

I wanted to raise this issue, but I am not worried because it's a fairly uncommon pattern.